request_handler.h

```diff
@@ -57,21 +56,27 @@ class Response {
         FILE_NOT_FOUND = 404
     };
 
-    void SetStatus(const ResponseCode response_code);
+    static std::unique_ptr<Response> Parse(const std::string& raw_response);
+
+    void SetStatus(int response_code);
     void AddHeader(const std::string& header_name, const std::string& header_value);
     void SetBody(const std::string& body);
     int GetStatus();
 
+    std::string body() const;
+
+    using Headers = std::vector<std::pair<std::string, std::string>>;
+    Headers headers() const;
+
     std::string ToString();
 
 private:
     std::string build_status_line();
     std::string build_header_string(const std::pair<std::string, std::string>& field);
     
     std::string response_body_;
-    ResponseCode status_code_;
+    int status_code_;
```

Comment:
The ProxyHandler needs to be able to handle arbitrary status codes
Options were: (1) make it an integer, (2) add every HTTP1.1 response code to the enumeration
In case (2), I can still imagine some websites returning custom response codes (it's allowed)